### PR TITLE
Debug plugin interaction

### PR DESCRIPTION
**Pull request by GrandSchtroumpf**
Activate manager from engine
Update canCall

### CODE REVIEW

File: projects/engine/src/engine/engine.ts [hunk 1]
```diff
@@ -14,9 +14,9 @@ export class Engine {
   constructor(private manager: BasePluginManager) {
     this.plugins['manager'] = manager
     // Activate the Engine & start listening on activation and deactivation
-    this.activatePlugin('manager').then(() => {
-      this.manager['engineActivatePlugin'] = (name: string) => this.activatePlugin(name)
-      this.manager['engineDeactivatePlugin'] = (name: string) => this.deactivatePlugin(name)
+    this.manager['engineActivatePlugin'] = (name: string) => this.activatePlugin(name)
```
Comment (yann300) [inline projects/engine/src/engine/engine.ts]: is this change needed or is it for cleaning the code?

Comment (GrandSchtroumpf) [reply projects/engine/src/engine/engine.ts]: It's needed

File: projects/engine/tests/engine.ts [hunk 2]
```diff
@@ -202,14 +202,38 @@ describe('Plugin interaction', () => {
...
+    expect(_currentRequest).toEqual({ from: 'fileManager', path: 'solidity' })
+    expect(updateReq).toHaveBeenCalledTimes(2)
   })
 
   test('Plugin can activate another', async () => {
     await manager.activatePlugin('solidity')
     await solidity.call('manager', 'activatePlugin', 'fileManager')
     const isActive = await manager.isActive('fileManager')
     expect(manager.activatePlugin).toHaveBeenCalledWith('fileManager')
-    expect(manager.canActivate).toHaveBeenCalledWith(solidity.profile, manager.profile)
```
Comment (yann300) [inline projects/engine/tests/engine.ts]: It looks like `manager.profile` from`expect(manager.canActivate).toHaveBeenCalledWith(solidity.profile, manager.profile)` was not expected cause the plugin that `solidity` try to activate is the `fileManager` . is this directly part of the fix?

Comment (yann300) [reply projects/engine/tests/engine.ts]: could you point me in the PR where the fix is?

Comment (GrandSchtroumpf) [reply projects/engine/tests/engine.ts]: The test was wrong and it's one reason why it was hard to detect the bug

Comment (GrandSchtroumpf) [reply projects/engine/tests/engine.ts]: > could you point me in the PR where the fix is?

I don't understand your question

Comment (yann300) [reply projects/engine/tests/engine.ts]: if you update the test here (which was before passing I guess), It means that you've added a change in this PR for making sure the test still pass.

Comment (GrandSchtroumpf) [reply projects/engine/tests/engine.ts]: The change is here : 
https://github.com/ethereum/remix-plugin/pull/171/files/78d3debe2e4c3e2b6bc78e3d4ed3e526fdc865b8#diff-c0e115918fb919bc94b8ec28711c6dd4R17

File: projects/engine/src/engine/engine.ts (hunk 1 above)
Comment (yann300) [reply projects/engine/src/engine/engine.ts]: could you describe why? I see that you are defining the callback before activation but how does that fix the issue (and need a change in the test)?

Comment (GrandSchtroumpf) [reply projects/engine/src/engine/engine.ts]: I was doing this.activatePlugin('manager'). Now I'm doing manager.activatePlugin('manager') which does more stuff (including running engine.activatePlugin('manager'))
